Alerting: Sort StateHistoryItem after fetch instead of on render.

joeblubaugh opened the pull request:
#47674 attempted to sort a read-only managed async array. This change
moves the sort logic to the fetch code so sort happens once on fetch, to
a mutable array, rather than trying on each render for an immutable
array.

Signed-off-by: Joe Blubaugh <joe.blubaugh@grafana.com>

**Special notes for your reviewer**:
Viewing state history is completely broken before this fix. Please prioritize this review.

joeblubaugh committed:
Alerting: Sort StateHistoryItem after fetch instead of on render.

attempted to sort a read-only managed async array. This change
moves the sort logic to the fetch code so sort happens once on fetch, to
a mutable array, rather than trying on each render for an immutable
array.

Signed-off-by: Joe Blubaugh <joe.blubaugh@grafana.com>

gillesdemey commented on public/app/features/alerting/unified/api/annotations.ts:
```diff
@@ -2,7 +2,38 @@ import { getBackendSrv } from '@grafana/runtime';
+      alertId,
+    })
+    .then((result) => {
+      result.sort(sortStateHistory);
```
You can just return the results of the `sort` function here :)

gillesdemey reviewed (approved):
LGTM :shipit:

joeblubaugh committed:
Use optional chaining in the Promise fullfillment function

The Promise can be fulfilled with an undefined result in some cases, so
use an optional chain to ensure we don't have a reference error.

Signed-off-by: Joe Blubaugh <joe.blubaugh@grafana.com>

joeblubaugh committed:
Make test mock return an actual promise instead of nothing.
